core/solver/batch_lower_trs.cpp

```diff
@@ -0,0 +1,161 @@
+/*******************************<GINKGO LICENSE>******************************
+Copyright (c) 2017-2022, the Ginkgo authors
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+1. Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in the
+documentation and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#include <ginkgo/core/solver/batch_lower_trs.hpp>
+
+
+#include <ginkgo/core/matrix/batch_csr.hpp>
+#include <ginkgo/core/matrix/batch_dense.hpp>
+#include <ginkgo/core/matrix/batch_ell.hpp>
+
+#include "core/matrix/batch_csr_kernels.hpp"
+#include "core/matrix/batch_dense_kernels.hpp"
+#include "core/solver/batch_lower_trs_kernels.hpp"
+
+
+namespace gko {
+namespace solver {
+namespace batch_lower_trs {
+
+
+GKO_REGISTER_OPERATION(apply, batch_lower_trs::apply);
+
+
+}  // namespace batch_lower_trs
+
+
+template <typename ValueType>
+std::unique_ptr<BatchLinOp> BatchLowerTrs<ValueType>::transpose() const
+{
+    return build()
+        .with_left_scaling_op(
+            share(as<BatchTransposable>(this->get_left_scaling_op())
+                      ->conj_transpose()))
+        .with_right_scaling_op(
+            share(as<BatchTransposable>(this->get_right_scaling_op())
+                      ->conj_transpose()))
+        .on(this->get_executor())
+        ->generate(share(
+            as<BatchTransposable>(this->get_system_matrix())->transpose()));
```

Comment:
Need to have `skip_sorting` as well here.

Reply:
done